### PR TITLE
RavenDB-5244 V4.0 New client : implement LoadStartingWith

### PR DESCRIPTION
**Pull request by iftahbe**
- Move transformers logic to helper class (shared between several operations)
- Fix LoadStartingWith to be case insensitive
- Fix parsing of strings from $values array
- Fix DocumentHandler in case of LoadStartingWith
- Enable skipped tests

### CODE REVIEW

**Comment by ayende** [comment]
Failing test
```
  SlowTests.MailingList.LoadAllStartingWith.LoadAllStartingWithShouldNotLoadDeletedDocs [FAIL]
    System.InvalidCastException : Unable to cast object of type 'Raven.Json.Linq.RavenJObject' to type 'Raven.Json.Linq.RavenJArray'.
    Stack Trace:
      C:\Work\ravendb-4.0\src\Raven.Client\Document\Batches\LazyStartsWithOperation.cs(90,0): at Raven.Client.Document.Batches.LazyStartsWithOperation`1.HandleResponse(GetResponse response)
      C:\Work\ravendb-4.0\src\Raven.Client\Document\DocumentSession.cs(991,0): at Raven.Client.Document.DocumentSession.ExecuteLazyOperationsSingleStep(ResponseTimeInformation responseTimeInformation)
      C:\Work\ravendb-4.0\src\Raven.Client\Document\DocumentSession.cs(943,0): at Raven.Client.Document.DocumentSession.ExecuteAllPendingLazyOperations()
      C:\Work\ravendb-4.0\src\Raven.Client\Document\DocumentSession.cs(879,0): at Raven.Client.Document.DocumentSession.<>c__DisplayClass73_0`1.<AddLazyOperation>b__0()
         at System.Lazy`1.CreateValue()
         at System.Lazy`1.LazyInitValue()
      C:\Work\ravendb-4.0\test\SlowTests\MailingList\LoadAllStartingWith.cs(45,0): at SlowTests.MailingList.LoadAllStartingWith.LoadAllStartingWithShouldNotLoadDeletedDocs()
```

File: src/Raven.Server/Documents/Handlers/DocumentHandler.cs [hunk 1]
```diff
@@ -119,7 +119,10 @@ private void GetDocuments(DocumentsOperationContext context, Transformer transfo
             }
             else if (HttpContext.Request.Query.ContainsKey("startsWith"))
             {
-                isLoadStartingWith = true;
+                // W.A. to distinguish between Lazily.LoadStartingWith of old client and LoadStartingWith of v4.0 new client
+                if (HttpContext.Request.Query.ContainsKey("ComingFromMultiGet") == false)
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/DocumentHandler.cs]: W.A. ? 

Why are you doing something like that? Why is there different behavior between the clients?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Handlers/DocumentHandler.cs (hunk 1 above)
Comment (iftahbe) [reply src/Raven.Server/Documents/Handlers/DocumentHandler.cs]: When requestring Lazily.LoadStartingWith in the old client, it's being sent as part of a multi-get (Post) operation, then the multi-get handler redirects it to the normal document handler (Get) and the request looks exactly (!) like a request from the new client but the result is in a different structure (object vs array). I needed a way to tell between them

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/DocumentHandler.cs]: Don't do it like that, change the old client to support the new way.
I don't want to have this kind of stuff in our codebase if we can avoid it, and we can do that if we change both.

Comment (iftahbe) [inline src/Raven.Server/Documents/Handlers/DocumentHandler.cs]: I mean it looks exactly like a request from LoadStartingWith in the new client

File: src/Raven.Server/Documents/Handlers/DocumentHandler.cs [hunk 2]
```diff
@@ -134,20 +137,29 @@ private void GetDocuments(DocumentsOperationContext context, Transformer transfo
 
             using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
             {
+                if (isLoadStartingWith)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/DocumentHandler.cs:140]: This introduces inconsistency. Endpoints that are returning documents return them in different JSON format depending of the option used. We should not go this way in our opinion.

Comment (ml054) [reply src/Raven.Server/Documents/Handlers/DocumentHandler.cs:140]: I will fix this as a part of: RavenDB-5796

**Pull request merged**